solr/core/src/java/org/apache/solr/handler/admin/BaseHandlerApiSupport.java

```diff
@@ -159,9 +159,7 @@ private Object getParams0(String param) {
             if (o == null) o = pathValues.get(param);
             if (o == null && useRequestParams) o = origParams.getParams(param);
             if (o instanceof List) {
-              @SuppressWarnings({"rawtypes"})
-              List l = (List) o;
-              return l.toArray(new String[l.size()]);
+              return ((List<?>) o).toArray();
```

Comment:
This change is very interesting but I'm struggling to convince myself that and how it won't affect the `instanceof String[]` condition outcomes in the callers?

Reply:
You're right, we need to use the toArray overload that accepts a type argument.